src/functions/formatedEnum.ts

```diff
@@ -0,0 +1,40 @@
+import { IFunction, IFunctionResult } from '../types';
+import { schema } from './schema';
+
+export const formatedEnum: IFunction = (targetVal, opts, paths, otherValues): void | IFunctionResult[] => {
+  const { enum: enumValues, ...innerSchema } = targetVal;
+  const schemaObject = { schema: innerSchema };
+
+  const incorrectValues: unknown[] = [];
+
+  // Maximum number of reported invalid enum values
+  const reportingThreshold = 3;
```

Comment:
The idea behind this is to not overburden the error message if the enum is pretty long and full of invalid values.

Optionally, this might be passed as a parameter to the function. Let me know if you think this add any value.

Reply:
Perhaps make it configurable?

Reply:
Ok. I'll make this configurable

Reply:
@P0lip Done. Does that fit your expectation?

Reply:
Yeah, looking good.